### ISSUE TITLE
How to know the iteration number inside custom eval function

### ISSUE THREAD

**Issue by jaguerrerod** [dmlc/xgboost#8143]
Hi, how to know (in R interface) the iteration number inside a custom eval function?
I have a very intensive custom eval function with a very small learning rate, so I'm only interested in check early stopping each 10 or 20 iterations and save time.

Thanks.

**Comment by trivialfis** [comment]
I think you can count the evaluation by a captured variable.

**Comment by jaguerrerod** [comment]
I found using <<- assignment of an iter counter inside custom eval works.
Using CV is necessary adjust the counter as eval is called for each fold.

**Comment by trivialfis** [comment]
Yup. Feel free to reopen if there are further questions.
